### ISSUE TITLE
SIlent crash after PC wake-up

### ISSUE THREAD

**Issue by johny-mnemonic** [keepassxreboot/keepassxc#5415]
## Overview
Approximately since updating to 2.5.x I am having the issue, that after I wake up the PC, KeePassXC seems to be running, but when I click on the tray icon, it silently crashes.

## Steps to Reproduce
1. Start KeePassXC and leave it runningin the background
2. Put your Win 10 PC into sleep
3. Wake up your PC and try to use KeePassXC

## Expected Behavior
Being able to use KeePassXC after PC wake-up

## Actual Behavior
App crashes without any error. Usually immediately once I click on it in try, sometimes only after I try to search.
It doesn't crash after every wake-up but often enough to be annoying (like 2 out of 3 times).

## Notes:
I am using portable version of KeePassXC stored in Dropbox to have same version running on all my 3 PCs.
Today it crashed on the one that has been in sleep over night as well as on the one that was awake. Seems that waking the PC does something to it's local version of db file, which then propagates (through Dropbox) to other PCs.
Today the PC that was online whole night was the first one to crash on me after I connected to it (RDP) from the one I have woken up. It even crashed with an error for the first time on me:
`KeePassXC: Windows - Application Error
The instruction at 0x0000000000745396 referenced memory at 0x0000000000000010. The memory could not be read. Click on OK to terminate the program`

I then tried to use the app on woken-up PC and it opened from tray just fine, but crashed immediately after I typed first letter into search box (this time it was silent crash without any error produced as usual)

Is there a way how to enable some (preferably debug) logging to see what is going on when the app silently crashes? The error produced today on the PC that was not even sleeping is the first error I have seen. Usually it just crashes without producing anything and I was unable to find any existing log or an option to enable one.

## Debug info
KeePassXC - Verze 2.6.1

Revize: 9a35bba

Qt 5.15.0
Ladící režim je vypnutý.

Operační systém: Windows 10 Version 2004
Architektura procesoru: x86_64
Jádro systému: winnt 10.0.19041

Operating System: Windows 10 Version 2004 PRO CZ

**Comment by droidmonkey** [comment]
Do not store the program in Dropbox. That is absolutely causing the crashes.

**Comment by johny-mnemonic** [comment]
I am really sorry, but I seriously doubt that. Unless it is caused by some new feature introduced to the app.
I am using KeePass and then KeePassXC this way at least since 2006 and never had any single issue with this.
Also I am using a lot of other apps this way (running from DropBox folder) without any issue.

Would you please be more specific why do you think this could cause any issue? 
The folder is in fact located on local disk, it's just that Dropbox synchronize changes done to it to other PCs. Do you started using some temporary files located in that folder that could be overwritten during runtime by two apps running from the same folder? This would be an issue even locally when starting two instances of the app if I am not missing something.
Dropbox shouldn't be locking files for reading, why would it do something like that?

**Comment by droidmonkey** [comment]
You can debug the situation using our snapshot builds: https://snapshot.keepassxc.org

**Comment by johny-mnemonic** [comment]
So I switched to the snapshot:
```
KeePassXC - Version 2.6.1-snapshot
Build Type: Snapshot
Revision: 6a35bbe
```
And since then I didn't have a single crash :-O
Does this log somewhere or the only purpose of these snapshot builds is to create stacktrace/dump in case it crashes?

**Comment by droidmonkey** [comment]
They are meant for crash dumps

**Comment by johny-mnemonic** [comment]
@droidmonkey Great. Would you please direct me where can I find the crash dump?
It definitely took much longer then expected, but after five days it finally crashed.

**Comment by droidmonkey** [comment]
https://helgeklein.com/blog/2018/10/creating-an-application-crash-dump/

But beware your dump file could contain sensitive information.

**Comment by johny-mnemonic** [comment]
OK. Found it.
Where can I send it? Considering it contains sensitive information I don't think attaching it here is a good idea.

**Comment by droidmonkey** [comment]
Team@keepassxc.org

**Comment by shoaloak** [comment]
I have the same behaviour on Wayland on Linux. After suspend keepassxc just hangs. Can I help fix this by e.g. sending a crash dump? If so, how does one do this correctly for Keepassxc?

**Comment by droidmonkey** [comment]
Debug info for your keepassxc please

**Comment by shoaloak** [comment]
I assume you mean `keepassxc --debug-info`?
```
KeePassXC - Version 2.6.2
Revision: e9b9582

Qt 5.15.2
Debugging mode is disabled.

Operating system: Arch Linux
CPU architecture: x86_64
Kernel: linux 5.9.14-arch1-1

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- Secret Service Integration

Cryptographic libraries:
 libgcrypt 1.8.7
```

Also, the last info KeePassXC ironically logs before crashing is:
`qt.qpa.wayland: Creating a fake screen in order for Qt not to crash`

The freeze/crash occurs with and without the Wayland QT environment variables:
```
QT_QPA_PLATFORM=wayland
QT_QPA_PLATFORMTHEME=qt4ct
```

**Comment by droidmonkey** [comment]
Seems like a Qt crash to me

**Comment by shoaloak** [comment]
I think you're right, setting `export QT_QPA_PLATFORM=xcb` solves it.
Guess it's a QT bug.

**Comment by LinuxOnTheDesktop** [comment]
Cf. #6582.

Also, and for a few versions of KeePassXC now, my Windows version of KeePassXC hangs occasionally. It does so - sometimes - in the following circumstance: after I usethe key combination to fill a form, I enter the master password to unlock the (previously locked) database. I suppose that I should open a new issue for that problem.

**Comment by droidmonkey** [comment]
Use our snapshot first because Auto-Type was completely overhauled on develop branch. https://snapshot.keepassxc.org

I'm closing this issue since it's Qt related

**Comment by LinuxOnTheDesktop** [comment]
I assume it is safe to use the development branch on one PC whilst syncing with devices that use the mainline branch.

**Comment by droidmonkey** [comment]
Yes they are completely compatible with each other except for keeshare

**Comment by LinuxOnTheDesktop** [comment]
Right. Thanks. I am trying the development branch. I will open a new issue if I encounter a crash or a hang (and will replace the development branch with the release version once a new release arrives).

**Comment by mbilas** [comment]
On my Windows 10 I have this problem very often. When my PC is locked after idle time, after unblock after entering a database password in KeePass, it doesn't respond. Version 2.6.6.
